cmd/queue/main.go

```diff
@@ -70,13 +70,14 @@ var (
 )
 
 type config struct {
-	ContainerConcurrency     int    `split_words:"true" required:"true"`
-	QueueServingPort         string `split_words:"true" required:"true"`
-	UserPort                 string `split_words:"true" required:"true"`
-	RevisionTimeoutSeconds   int    `split_words:"true" required:"true"`
-	ServingReadinessProbe    string `split_words:"true"` // optional
-	EnableProfiling          bool   `split_words:"true"` // optional
-	EnableHTTP2AutoDetection bool   `split_words:"true"` // optional
+	ContainerConcurrency              int    `split_words:"true" required:"true"`
+	QueueServingPort                  string `split_words:"true" required:"true"`
+	UserPort                          string `split_words:"true" required:"true"`
+	RevisionTimeoutSeconds            int    `split_words:"true" required:"true"`
+	RevisionMaxDurationTimeoutSeconds int    `split_words:"true" required:"true"`
```

Comment:
Do we have to worry about upgrades not having this env var, e.g. if the defaults ConfigMap with the new QP version updates before the new controller code is deployed? (I guess maybe not since the upgrade tests didn't fail 🤔)

Reply:
I dont set any defaults at the revision level to avoid such issues. If the env var does not exist then the new QP code will [pick zero](https://github.com/knative/serving/blob/c08ea02486a9b27a9a6ef55d09d15babd3285d48/pkg/reconciler/revision/resources/queue.go#L194) as the value for the timeout and will ignore it later on.

Reply:
but this env var is set as required here, so won't envconfig panic on startup if this version of queue proxy rolls out before the corresponding code change that adds the env var?

Reply:
Correct my intention was to make it optional, good point copy past left over.